### ISSUE TITLE
Box plots with custom fill colors

### ISSUE THREAD

**Issue by mickaellalande** [lukelbd/proplot#217]
### Description

I would like to make boxplot with custom colors as in this example from matplotlib: https://matplotlib.org/3.1.1/gallery/statistics/boxplot_color.html, however, it looks like we don't have access to `.set_facecolor` from a `bplot['boxes']` with proplot and/or the argument `fillcolor` doesn't take list.

### Steps to reproduce

**Expected behavior**: 

Here is the matplotlib example (from the link above, a bit modified):

```python
import matplotlib.pyplot as plt
import numpy as np

# Random test data
np.random.seed(19680801)
all_data = [np.random.normal(0, std, size=100) for std in range(1, 4)]
labels = ['x1', 'x2', 'x3']

fig, ax = plt.subplots()

# rectangular box plot
bplot = ax.boxplot(all_data, patch_artist=True, labels=labels)  
ax.set_title('Rectangular box plot')

# fill with colors
colors = ['pink', 'lightblue', 'lightgreen']
for patch, color in zip(bplot['boxes'], colors):
    patch.set_facecolor(color)

# adding horizontal grid lines
ax.yaxis.grid(True)
ax.set_xlabel('Three separate samples')
ax.set_ylabel('Observed values')

plt.show()
```
![Capture d’écran du 2020-08-13 11-43-06](https://user-images.githubusercontent.com/20254164/90119841-5faa1a80-dd5a-11ea-8b62-7cda02bedd38.png)

**Actual behavior**:

Same with Proplot, I've just transformed the data in pandas DataFrame (I guess it's needed for Proplot?):

```python
import proplot as plot
import numpy as np
import pandas as pd

# Random test data
np.random.seed(19680801)
all_data = [np.random.normal(0, std, size=100) for std in range(1, 4)]
labels = ['x1', 'x2', 'x3']

# Convert to pandas dataframe
df = pd.DataFrame(None, columns=pd.Index(labels, name='xlabel'))
for i, data in enumerate(all_data):
    df[labels[i]] = data

fig, ax = plot.subplots()

# rectangular box plot
bplot = ax.boxplot(df)  
# bplot = ax.boxplot(df, fillcolor=['pink', 'lightblue', 'lightgreen'])  
ax.set_title('Rectangular box plot')

#####################################################
# fill with colors
colors = ['pink', 'lightblue', 'lightgreen']
for patch, color in zip(bplot['boxes'], colors):
    patch.set_facecolor(color)
#####################################################

# adding horizontal grid lines
ax.format(grid=True, xlabel='Three separate samples', ylabel='Observed values')
```

Returns:

`AttributeError: 'Line2D' object has no attribute 'set_facecolor'
`

So I don't know if there is another way, I tried to pass a list to the fill `fillcolor` option from Proplot (as in this example: https://proplot.readthedocs.io/en/latest/1dplots.html#Box-plots-and-violin-plots), however it didn't work too. 

Thanks before for the help if I missed something.

### Proplot version

0.6.4

**Comment by mickaellalande** [comment]
I finally found a solution by checking at the code (https://proplot.readthedocs.io/en/latest/_modules/proplot/axes/plot.html#boxplot_wrapper):

I have just changed the part from my code above:

```python
# fill with colors
colors = ['pink', 'lightblue', 'lightgreen']
for patch, color in zip(bplot['boxes'], colors):
    patch.set_facecolor(color)
```

by:

```python
import matplotlib.patches as mpatches
# fill with colors
colors = ['pink', 'lightblue', 'lightgreen']
for artist, color in zip(bplot['boxes'], colors):
    patch = mpatches.PathPatch(artist.get_path(), color=color)
    ax.add_artist(patch)
```

And it works! :smile: 

I guess it would be nice to have another argument, like for example: `fillcolors` (and others) or take into account the lists!

**Comment by lukelbd** [comment]
This was solved resolved by #218 (thanks again!)
